### ISSUE TITLE
Optimization to urweb daemon: Cache generated code when Ur source doesn't change

### ISSUE THREAD

**Issue by FrigoEU** [urweb/urweb#133]
I understood that caching generated code on a per-module basis is very difficult. Some thing that I think would be easier and would help me out a lot would be to cache the full code generated, and reuse it on a rebuild when non of the Ur code has changed. 

This would help tremendously when working on CSS: I constantly have to go through rebuilds (+- 15 secs now) when I'm only changing CSS. I try to alleviate the problem by doing as much design as possible in Chrome Dev Tools, but it would still be a huge improvement.

**Comment by achlipala** [comment]
Yeah, if the Ur/Web code has stayed the same, it should be pretty easy to cache the expensive compiler phases' results.  Let me think about this one.... I sure wouldn't mind if someone else submitted a patch for it, but I should be able to get to it soon-ish.

**Comment by achlipala** [comment]
I played around a bit with solutions, with an eye toward maximal simplicity, and I realized some complications with `.urp` settings influencing code generation.  Would you be satisfied with reuse of compiler work only when all `.ur*` files are unchanged?  Are you mostly interested in rerunning to pick up new content of files linked into a binary with the `file` directive in a `.urp` file?

**Comment by FrigoEU** [comment]
Yes, totally fine with your suggestion. My use case is pretty much
exclusively css changes.

**Comment by achlipala** [comment]
At the moment this task has fallen off of my "soon-ish" list, so I'll appreciate if someone else wants to give it a try.
